docs/book/beyond_basics/deploying_controller.md

```diff
@@ -0,0 +1,84 @@
+# Deploy the controller-manager in a Kubernetes cluster
+
+Deploying the controller to a Kubernetes cluster involves following steps:
+ - Building the docker image
+ - Pushing the docker image to the container registry
+ - Customizing the deployment manifests
+ - Applying the manifests to deploy in the cluster
+
+Kubebuilder generated `Makefile` supports all the above steps.
+
+{% panel style="info", title="Prerequisites" %}
+Kubebuilder generated `Makefile` uses [Kustomize](https://github.com/kubernetes-sigs/kustomize) for customizing the manifests
+before deploying to the kubernetes cluster. Follow the [instructions](https://github.com/kubernetes-sigs/kustomize/blob/master/INSTALL.md) to install `Kustomize` and
+ensure that is available in the PATH. Note that Kubebuilder requires `Kustomize` version `1.0.4` or higher for deploy to work.
+
+```bash
+opsys=linux  # or darwin, or windows
+curl -s https://api.github.com/repos/kubernetes-sigs/kustomize/releases/latest |\
+  grep browser_download |\
+  grep $opsys |\
+  cut -d '"' -f 4 |\
+  xargs curl -O -L
+mv kustomize_*_${opsys}_amd64 kustomize
+chmod u+x kustomize
+```
+
+The yaml configuration for the Manager is automatically created under
+`config/manager`.
+{% endpanel %}
+
+#### Building the docker image
+
+`Makefile` has a target `docker-build` to build the docker image for the
+controller manager. The target supports `IMG` variable.
+
+```bash
+# build the docker image 
+make docker-build IMG=<image-name>
```

Comment:
IMO we should also say that the target will pick it from the environment variable if the variable is not provided.
```
export IMG=<image-name>
make docker-build
```